packages/rds/src/styles/default.scss

```diff
@@ -0,0 +1,99 @@
+/**
+ * Default CSS styles. This file is similar to a "reset" stylesheet and
+ * represents base styles for consistency across all RDS components.
+ *
+ * This stylesheet can be imported in two different ways:
+ *
+ * 1) Import within `BaseLayout.tsx`:
+ *
+ * ```
+ * import '@blinkk/rds/styles/default.scss';
+ * ```
+ *
+ * Note: Cannot configure the styles using this method. See (2) if you need to
+ * configure the CSS output.
+ *
+ * 2) Import within a SCSS file, e.g. `global.scss`:
+ *
+ * ```
+ * @use '@blinkk/rds/styles/default.scss';
+ * ```
+ *
+ * To change any configurations, use the `@use ... with` syntax, e.g.:
+ *
+ * ```
+ * @use '@blinkk/rds/styles/default.scss' with (
+ *   $disable-prefers-reduced-motion: true;
+ * );
+ * ```
+ */
+
+$disable-smooth-scroll: false !default;
+$disable-prefers-reduced-motion: false !default;
+
+@layer rds.base {
+  //  Use a more intuitive box-sizing strategy.
+  *,
+  *::before,
+  *::after {
+    box-sizing: border-box;
+  }
+
+  // Reset margins.
+  body {
+    margin: 0;
+  }
+
+  // The default `display: inline` property of img and video causes extra
+  // vertical space within a parent container. `display: block` is generally
+  // more intuitive. Also set a max-width and auto height based on the
+  // width and height attributes of the element.
+  img,
+  svg,
+  video {
+    display: block;
+    max-width: 100%;
+    height: auto;
+  }
+
+  // Inherit font for form elements.
+  input,
+  button,
+  textarea,
+  select {
+    font: inherit;
+  }
+
+  // Set smooth scrolling by default. The `:focus-within` helps with cmd+f
+  // functionality.
+  // https://css-tricks.com/fixing-smooth-scrolling-with-find-on-page/
+  @if not($disable-smooth-scroll) {
+    html:focus-within {
+      scroll-behavior: smooth;
+    }
+    html {
+      --scroll-padding-top: var(--header-height, 0px);
```

Comment:
We don't need to handle this right now but I'm wondering if we want to adopt a prefixing strategy for our CSS vars, e.g. `--rds-scroll-padding-top`.

I look at the styles for https://github.com/ in incognito to get the marketing page (GitHub's been a long time user of CSS vars and tokens), and they do NOT use such prefixes. Stripe doesn't either. So maybe we don't -- just wanted to bubble it up for discussion.

Reply:
I gave it some thought, I think generally when I see a prefix like `--rds-` I'd see it as an encapsulated configuration variable that's either used internally by the component or is exposed by the component as an API. So like, if for example I have an `<rds-button>` component, maybe the rds-button would have a CSS variable like `--rds-button-icon-size` to provide configuration options to the user.

For general-purpose variables that can be used throughout the site, I think I prefer non-rds-prefixed variables. Feels a little cleaner and doesn't feel like I'm reaching in and modifying something that I shouldn't be modifying.

I took a look at your github.com example and I really liked how clean some of their base CSS variables are, e.g.

```
--h1-size: 2rem;
--breakpoint-small: 34rem;
```

That said, they're not super consistent and in some cases they do prefix with the word `base` (I assume "base" is the name of their design system.), e.g.:

```
--base-text-weight-light: 300;
```

For now, my recommendation is to prefix based on functionality, and reserve the `--rds-` prefix for configuring `<rds-*>` custom elements.